Does not build on 64 bit

Welcome to the Haiku shell.

~/HaikuArchives/Tipster> make
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/Shuffle.d" source/Shuffle.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/TipsterText.d" source/TipsterText.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/Tipster.d" source/Tipster.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/MainWindow.d" source/MainWindow.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/App.d" source/App.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -Iobjects.x86_64-cc5-release/  -Isource/ -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/AboutTipster.d" source/AboutTipster.cpp
g++ -c source/AboutTipster.cpp -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquotesource/   -O3    -o "objects.x86_64-cc5-release/AboutTipster.o"
g++ -c source/App.cpp -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquotesource/   -O3    -o "objects.x86_64-cc5-release/App.o"
g++ -c source/MainWindow.cpp -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquotesource/   -O3    -o "objects.x86_64-cc5-release/MainWindow.o"
g++ -c source/Tipster.cpp -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquotesource/   -O3    -o "objects.x86_64-cc5-release/Tipster.o"
source/Tipster.cpp: In member function 'void Tipster::UpdateTip()':
source/Tipster.cpp:400:15: error: invalid array assignment
   fRandomSeq1 = fRandomSeq2;
               ^
/boot/system/develop/etc/makefile-engine:291: recipe for target 'objects.x86_64-cc5-release/Tipster.o' failed
make: *** [objects.x86_64-cc5-release/Tipster.o] Error 1
~/HaikuArchives/Tipster>

--- a/source/Shuffle.cpp
+++ b/source/Shuffle.cpp
@@ -8,11 +8,11 @@
 // don't use random_shuffle deprecated in the future...
 
 void 
-Randomize(std::vector<int> &arr, int n)
+Randomize(std::vector<int> &arr)
 {
 	srand ( time(NULL) );
 
-	for(int i = n - 1; i > 0; i--) {
+	for(int i = arr.size() - 1; i > 0; i--) {
 		int j = rand() % (i+1);
 		std::swap(arr[i], arr[j]);
 	}
@@ -24,5 +24,5 @@ CreateRandomSeq(std::vector<int> &arr, int len)
 {
 	for(int i = 0; i < len; ++i)
 		arr.push_back(i);
-	Randomize(arr, len);
+	Randomize(arr);
 }

--- a/source/Shuffle.cpp
+++ b/source/Shuffle.cpp
@@ -5,31 +5,24 @@
 #include "Shuffle.h"
 
 
-void 
-Swap(int* a, int* b)
-{
-	int temp = *a;
-	*a = *b;
-	*b = temp;
-}
-
+// don't use random_shuffle deprecated in the future...
 
 void 
-Randomize(int* arr, int n)
+Randomize(std::vector<int> &arr, int n)
 {
 	srand ( time(NULL) );
 
 	for(int i = n - 1; i > 0; i--) {
 		int j = rand() % (i+1);
-		Swap(&arr[i], &arr[j]);
+		std::swap(arr[i], arr[j]);
 	}
 }
 
 
 void
-CreateRandomSeq(int* arr, int len)
+CreateRandomSeq(std::vector<int> &arr, int len)
 {
 	for(int i = 0; i < len; ++i)
-		arr[i] = i;
+		arr.push_back(i);
 	Randomize(arr, len);
 }

--- a/source/Shuffle.h
+++ b/source/Shuffle.h
@@ -8,11 +8,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <time.h>
+#include <vector>
 
-
-void Swap(int* a, int* b);
 void Randomize(int* arr, int n);
-void CreateRandomSeq(int* arr, int len);
+void CreateRandomSeq(std::vector<int> &arr, int len);
 
 #endif
 

--- a/source/Shuffle.h
+++ b/source/Shuffle.h
@@ -10,7 +10,7 @@
 #include <time.h>
 #include <vector>
 
-void Randomize(int* arr, int n);
+void Randomize(std::vector<int> &arr);
 void CreateRandomSeq(std::vector<int> &arr, int len);
 
 #endif

--- a/source/Tipster.h
+++ b/source/Tipster.h
@@ -15,6 +15,7 @@
 #include <StringList.h>
 #include <TextView.h>
 
+#include <vector>
 
 class Tipster : public BGroupView
 {
@@ -55,8 +56,8 @@ private:
 	BStringList 		fTipsList;
 	int32			fTipsLength;
 	int32			fTipIndex;
-	int 			fRandomSeq1[50];
-	int 			fRandomSeq2[50];
+	std::vector<int> fRandomSeq1;
+	std::vector<int> fRandomSeq2;
 	
 	BString*		fPreviousTip;
 	BString*		fCurrentTip;

